scri/flux.py

```diff
@@ -437,12 +429,298 @@ def angular_momentum_flux(h, hdot=None):
+
+        cg1 = np.sqrt(2) * CG(l1, m1, l2, m2, l3, m3)
+        cg2 = CG(l1, -1.0, l2, -s, l3, -1 - s)
+
+        return np.sqrt((2.0 * l1 + 1.0) * (2.0 * l2 + 1.0) / (4.0 * np.pi * (2.0 * l3 + 1))) * cg1 * cg2
+
+    for ell in range(ell_min, ell_max + 1):
+        ellp_min = max(ell_min, ell - 1)
+        ellp_max = min(ell_max, ell + 1)
+        for ellp in range(ellp_min, ellp_max + 1):
+            for m in range(-ell, ell + 1):
+                mp = round(m + 1 * sign)
+                if (mp < -ellp) or (mp > ellp):
+                    continue
+                yield (ellp, mp, ell, m, (prefac * mat_el_eth_chi(s, ellp, mp, 1.0, sign, ell, m)))
+
+
+eth_chi_plus = functools.partial(eth_chi_plusminus, sign=+1)
+eth_chi_minus = functools.partial(eth_chi_plusminus, sign=-1)
+eth_chi_plus.__doc__ = eth_chi_plusminus.__doc__
+eth_chi_minus.__doc__ = eth_chi_plusminus.__doc__
+
+
+@swsh_indices_to_matrix_indices
+def ethbar_chi_plusminus(ell_min, ell_max, sign, s=-2):
+    """Compute the <h|\\eth \\chi|\\eth N> matrix element depending on the sign.
+    Plus and Minus sign corresponds to the value taken by m = ±1.
+    Because \\chi is l = 1 spherical harmonic function.
+
+    The analytical expression is:
+    <h|\\eth \\chi|\\eth N> = \\sqrt{\frac{(2*l1+1)*(2*l2+1)}{4*\\pi*(2*l3+1})}*\\sqrt{2}*<l1,1,l2,-1-s|l3,-s>*<l1,m1,l2,m2|l3,m3>
+
+    We use `mat_el_ethbar_chi` to compute the matrix elements.
+    The operator has definite m = ±1, we only have mp == m ± 1 nonvanishing in the matrix elements.
+    """
+
+    if (sign != 1) and (sign != -1):
+        raise ValueError("sign must be either 1 or -1 in h_ethbar_chi_eth_N_plusminus")
+
+    prefac = -1.0 * sign * np.sqrt(8.0 * np.pi / 3.0)
+
+    def mat_el_ethbar_chi(s, l3, m3, l1, m1, l2, m2):
+
+        cg1 = np.sqrt(2) * CG(l1, m1, l2, m2, l3, m3)
+        cg2 = CG(l1, 1.0, l2, -1 - s, l3, -s)
+
+        return np.sqrt((2.0 * l1 + 1.0) * (2.0 * l2 + 1.0) / (4.0 * np.pi * (2.0 * l3 + 1))) * cg1 * cg2
+
+    for ell in range(ell_min, ell_max + 1):
+        ellp_min = max(ell_min, ell - 1)
+        ellp_max = min(ell_max, ell + 1)
+        for ellp in range(ellp_min, ellp_max + 1):
+            for m in range(-ell, ell + 1):
+                mp = round(m + 1 * sign)
+                if (mp < -ellp) or (mp > ellp):
+                    continue
+                yield (ellp, mp, ell, m, (prefac * mat_el_ethbar_chi(s, ellp, mp, 1.0, sign, ell, m)))
+
+
+ethbar_chi_plus = functools.partial(ethbar_chi_plusminus, sign=+1)
+ethbar_chi_minus = functools.partial(ethbar_chi_plusminus, sign=-1)
+ethbar_chi_plus.__doc__ = ethbar_chi_plusminus.__doc__
+ethbar_chi_minus.__doc__ = ethbar_chi_plusminus.__doc__
+
+
+def boost_flux(h, hdot=None):
+    """p_z, p_plusminus and the matrix elements defined above will be used here.
+    matrix_expectation_value is used to calcuate the component of the flux.
+    The spin value taken by the matrix element is specified for each term.
+    """
```

Comment:
Since people will be calling the help text of this function independently could this help text be a bit less code-specific? Within a python interpreter the text `p_z....defined above` will have no meaning to a user. See the help text of `energy_flux` for an example.

Reply:
Okay, I will try to change it. There is no expression for boost flux in any paper hence, I couldn't refer to any.